sql/stats: TestReschedulingOnConcurrentCreateStatsError failed

sql/stats.TestReschedulingOnConcurrentCreateStatsError [failed](https://mesolite.cluster.engflow.com/invocations/default/67f9edc0-4669-4094-87ff-1d6d53b74097?testReportRun=3&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsL3N0YXRzOnN0YXRzX3Rlc3Q=) on master @ [e99199606f52e553ab0a5198893ad44215d4f98c](https://github.com/cockroachdb/cockroach/commits/e99199606f52e553ab0a5198893ad44215d4f98c):


```
=== RUN   TestReschedulingOnConcurrentCreateStatsError
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestReschedulingOnConcurrentCreateStatsError105699629
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    create_stats_job_test.go:929: <nil>
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestReschedulingOnConcurrentCreateStatsError105699629
--- FAIL: TestReschedulingOnConcurrentCreateStatsError (7.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReschedulingOnConcurrentCreateStatsError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62426

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -854,8 +854,6 @@ func TestReschedulingOnConcurrentCreateStatsError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderRace(t, "the test is too sensitive to overload")
-
 	defer func(oldRefreshInterval, oldAsOf time.Duration) {
 		stats.DefaultRefreshInterval = oldRefreshInterval
 		stats.DefaultAsOfTime = oldAsOf
@@ -901,17 +899,13 @@ func TestReschedulingOnConcurrentCreateStatsError(t *testing.T) {
 	// Lower the concurrency limit to allow at most one auto full stats job.
 	sqlDB.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_full_concurrency_limit = 1`)
 
-	// Create two tables. Notably auto stats are enabled on 't2'.
 	sqlDB.Exec(t, `CREATE DATABASE d`)
 	sqlDB.Exec(t, `CREATE TABLE d.t (x INT PRIMARY KEY)`)
-	sqlDB.Exec(t, `CREATE TABLE d.t2 (x INT PRIMARY KEY) WITH (sql_stats_automatic_collection_enabled = true)`)
 	sqlDB.Exec(t, `INSERT INTO d.t SELECT generate_series(1,1000)`)
 
-	descT2 := desctestutils.TestingGetTableDescriptor(s.DB(), s.Codec(), "d", "public", "t2")
-
 	// Block the next stats collection on the table 't'.
-	var tID, t2ID descpb.ID
-	sqlDB.QueryRow(t, `SELECT 'd.t'::regclass::int, 'd.t2'::regclass::int`).Scan(&tID, &t2ID)
+	var tID descpb.ID
+	sqlDB.QueryRow(t, `SELECT 'd.t'::regclass::int`).Scan(&tID)
 	setTableID(tID)
 
 	// Start an auto full stat job and wait until it's done one scan. This will
@@ -926,11 +920,22 @@ func TestReschedulingOnConcurrentCreateStatsError(t *testing.T) {
 	select {
 	case allowRequest <- struct{}{}:
 	case err := <-backgroundAutoFullStatErrCh:
-		t.Fatal(err)
+		if err != nil {
+			t.Fatal(err)
+		} else {
+			t.Fatal("query unexpectedly finished")
+		}
 	}
 
-	// Don't block the other stats jobs.
+	// Don't block other stats jobs.
 	setTableID(descpb.InvalidID)
+
+	// Now create the second table on which we'll test the behavior of the auto
+	// stats refresher.
+	sqlDB.Exec(t, `CREATE TABLE d.t2 (x INT PRIMARY KEY) WITH (sql_stats_automatic_collection_enabled = true)`)
+	var t2ID descpb.ID
+	sqlDB.QueryRow(t, `SELECT 'd.t2'::regclass::int`).Scan(&t2ID)
+	descT2 := desctestutils.TestingGetTableDescriptor(s.DB(), s.Codec(), "d", "public", "t2")
 
 	// This should trigger a refresh on 't2' which should hit the
 	// ConcurrentCreateStatsError (due to global concurrency limit being

